x-pack/platform/plugins/shared/cases/server/client/cases/utils.test.ts

```diff
@@ -1073,6 +1075,183 @@ describe('utils', () => {
     });
   });
 
+  describe('getInProgressInfoForUpdate', () => {
+    const date = '2021-02-03T17:41:26.108Z';
+
+    it('returns the correct in_progress_at info when the case is marked as in-progress', async () => {
+      expect(
+        getInProgressInfoForUpdate({ status: CaseStatuses['in-progress'], updatedAt: date })
+      ).toEqual({
+        in_progress_at: date,
+      });
+    });
+
+    it('should not reset in_progress_at when the case is reopened', async () => {
+      expect(
+        getInProgressInfoForUpdate({ status: CaseStatuses.open, updatedAt: date })
+      ).toBeUndefined();
+    });
+
+    it('returns undefined if the status is not provided', async () => {
+      expect(getInProgressInfoForUpdate({ updatedAt: date })).toBeUndefined();
+    });
+  });
+
+  describe('getTimingMetricsForUpdate', () => {
+    const createdAt = '2021-11-23T19:00:00Z';
+    const inProgressAt = '2021-11-23T19:00:10Z';
+    const updatedAt = '2021-11-23T19:00:20Z';
+
+    it('should return the correct time_to_acknowledge when the case is marked as in-progress', () => {
+      expect(
+        getTimingMetricsForUpdate({
+          status: CaseStatuses['in-progress'],
+          createdAt,
+          updatedAt,
+        })
+      ).toEqual({
+        time_to_acknowledge: 20,
+        time_to_investigate: null,
+        time_to_resolve: null,
+      });
+    });
+
+    it('should not overwrite time_to_acknowledge', () => {
+      expect(
+        getTimingMetricsForUpdate({
+          status: CaseStatuses['in-progress'],
+          createdAt,
+          inProgressAt,
+          updatedAt,
+          timeToAcknowledge: 1,
+        })
+      ).toBeUndefined();
+    });
+
+    it('should return the correct time_to_investigate and time_to_resolve when the case transitions from in-progress to closed', () => {
+      expect(
+        getTimingMetricsForUpdate({
+          status: CaseStatuses.closed,
+          createdAt,
+          inProgressAt,
+          updatedAt,
+          timeToAcknowledge: 5,
+        })
+      ).toEqual({
+        time_to_investigate: 10,
+        time_to_resolve: 20,
+      });
+    });
+
+    it('should return the correct time_to_acknowledge, time_to_investigate and time_to_resolve when the case transitions from open to closed', () => {
+      expect(
+        getTimingMetricsForUpdate({
+          status: CaseStatuses.closed,
+          createdAt,
+          updatedAt,
+        })
+      ).toEqual({
+        time_to_acknowledge: 20,
```

Comment:
Shouldn't this be `10`? Is it because `inProgressAt` is not defined?

Reply:
Yes, when transitioning directly from open to closed without passing through in-progress I used the close date as the in-progress date, see [here](https://github.com/elastic/kibana/blob/b8effb9e26e45e81f660fd1467d8032784c377bc/x-pack/platform/plugins/shared/cases/server/client/cases/utils.ts#L482)